crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1001 @@
+};
+
+/// Stores [`ScheduleLabel`]-[`Schedule`] pairs.
+#[derive(Default, Resource)]
+pub struct Schedules {
+    inner: HashMap<BoxedScheduleLabel, Option<Schedule>>,
+}
+
+impl Schedules {
+    pub fn new() -> Self {
+        Self {
+            inner: HashMap::new(),
+        }
+    }
+
+    /// Insert a new labeled schedule into the map.
+    ///
+    /// # Errors
+    ///
+    /// TODO
+    pub fn insert(
+        &mut self,
+        label: impl ScheduleLabel,
+        schedule: Schedule,
+    ) -> Result<(), InsertionError> {
+        let label = label.dyn_clone();
+        if self.inner.contains_key(&label) {
+            return Err(InsertionError::AlreadyExists(label));
+        }
+        self.inner.insert(label, Some(schedule));
+        Ok(())
+    }
+
+    /// Returns the schedule corresponding to the label.
+    ///
+    /// # Errors
+    ///
+    /// TODO
+    pub(crate) fn check_out(
+        &mut self,
+        label: &dyn ScheduleLabel,
+    ) -> Result<Schedule, ExtractionError> {
+        let label = label.dyn_clone();
+        match self.inner.get_mut(&label) {
+            Some(container) => match container.take() {
+                Some(schedule) => Ok(schedule),
+                None => Err(ExtractionError::AlreadyExtracted(label)),
+            },
+            None => Err(ExtractionError::Unknown(label)),
+        }
+    }
+
+    /// Re-inserts the schedule corresponding to the label into the map.
+    ///
+    /// The schedule must have been previously extracted using [`check_out`](#method.check_out).
+    ///
+    /// # Errors
+    ///
+    /// TODO
+    pub(crate) fn check_in(
+        &mut self,
+        label: &dyn ScheduleLabel,
+        schedule: Schedule,
+    ) -> Result<(), ExtractionError> {
+        let label = label.dyn_clone();
+        match self.inner.get_mut(&label) {
+            Some(container) => match container.take() {
+                Some(_) => Err(ExtractionError::NotExtracted(label)),
+                None => {
+                    *container = Some(schedule);
+                    Ok(())
+                }
+            },
+            None => Err(ExtractionError::Unknown(label)),
+        }
+    }
+
+    /// Iterates all system change ticks and clamps any older than [`MAX_CHANGE_AGE`](crate::change_detection::MAX_CHANGE_AGE).
+    /// This prevents overflow and thus prevents false positives.
+    ///
+    /// **Note:** Does nothing if the [`World`] counter has not been incremented at least [`CHECK_TICK_THRESHOLD`](crate::change_detection::CHECK_TICK_THRESHOLD)
+    /// times since the previous pass.
+    pub(crate) fn check_change_ticks(&mut self, change_tick: u32, last_check_tick: u32) {
+        #[cfg(feature = "trace")]
+        let _span = bevy_utils::tracing::info_span!("check ticks").entered();
+        for schedule in self.inner.values_mut().flatten() {
+            schedule.check_change_ticks(change_tick, last_check_tick);
+        }
+    }
+}
+
+/// TBD
+pub struct Schedule {
```

Comment:
```suggestion
/// A collection of systems, and the metadata needed to order and run them.
pub struct Schedule {
```

Reply:
I tweaked the wording a bit, but this was good.